protocol/x/clob/e2e/order_matches_test.go

```diff
@@ -0,0 +1,144 @@
+package clob_test
+
+import (
+	"testing"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/app"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	clobtestutils "github.com/dydxprotocol/v4-chain/protocol/testutil/clob"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	testtx "github.com/dydxprotocol/v4-chain/protocol/testutil/tx"
+	clobtypes "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+	feetiertypes "github.com/dydxprotocol/v4-chain/protocol/x/feetiers/types"
+	satypes "github.com/dydxprotocol/v4-chain/protocol/x/subaccounts/types"
+	"github.com/stretchr/testify/require"
+)
+
+// BlockAdvancement holds orders and matches to be placed in a block. Using this struct and building
+// the ops queue with the getOperationsQueue helper function allows us to build the operations queue
+// without going through CheckTx and, therefore, not affect the local memclob state. This also allows us to propose
+// an invalid set of operations that an honest validator would not generate.
+type BlockAdvancement struct {
+	ordersAndMatches []interface{} // should hold Order and OperationRaw. OperationRaw are assumed to be matches.
+	expectedError    string
+}
```

Comment:
Can we add this struct definition in a testutil file? It seems like this could be used across multiple types of tests.

Reply:
Will do